### PR TITLE
feat: identity federation - OpenID connect

### PR DESCRIPTION
**Pull request by tinovyatkin**
Implements #477 except `reverse` functionality that will be implemented and documented in a dedicated PR.

### CODE REVIEW

File: extensions/exposition/documentation/components.md [hunk 1]
```diff
@@ -89,9 +89,20 @@ password?: string
...
+
+The `identity.federation` component manages OpenID Connect federated identities.
+
+Both implicit identities creation and forced [identity inception](./identity.md) are supported
+as in case with basic credentials. `principal` is also working in the same way.
+
+The configuration schema alongside it's default values is described in the [component manifest](../components/identity.federation/manifest.toa.yaml).
+
+No federated tokens are accepted by default until at least one of `allowed_issuers` or `acceptable_audience` is explicitly defined in the configuration.
+
```
Comment (temich) [inline extensions/exposition/documentation/components.md:114]: Add configuration example.

File: extensions/exposition/components/identity.federation/manifest.toa.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        description: The value of `sub` of an identity token that will be assigned the `system` Role
+        type: string
+    additionalProperties: false
+
+exposition:
+  isolated: true
+  /:
+    anonymous: true
+    POST: incept
+  /:id:
```
Comment (temich) [inline extensions/exposition/components/identity.federation/manifest.toa.yaml]: Remove this resource.

File: extensions/exposition/components/identity.federation/manifest.toa.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    input: string
+    output:
+      identity:
+        id: string
+
+configuration:
+  schema:
+    type: object
+    properties:
+      allowed_issuers:
```
Comment (temich) [inline extensions/exposition/components/identity.federation/manifest.toa.yaml]:
```yaml
trust:
  - iss*: a
    aud: b
```

File: libraries/http/source/Captures.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import * as assert from 'node:assert'
+import { binding } from 'cucumber-tsflow'
```
Comment (temich) [inline libraries/http/source/Captures.ts]: Remove this import, add `@binding()` where cucumber is used.

**Review by temich — CHANGES_REQUESTED** [review]

File: extensions/exposition/documentation/components.md (hunk 1 above)
Comment (tinovyatkin) [reply extensions/exposition/documentation/components.md:114]: Added GitHub federation example that allows two users (coincidentally @temich  and @tinovyatkin )

Comment (tinovyatkin) [reply extensions/exposition/documentation/components.md:114]: see [this commit](https://github.com/toa-io/toa/pull/486/commits/301a7e83e86c21307cd2a7e737f5412f955b543f)

**Review by temich — APPROVED** [review]

**Pull request merged**